projects/packages/forms/src/abilities/class-forms-abilities.php

```diff
@@ -0,0 +1,455 @@
+						),
+						'search'    => array(
+							'type'        => 'string',
+							'description' => __( 'Search within submission content and sender info.', 'jetpack-forms' ),
+						),
+						'before'    => array(
+							'type'        => 'string',
+							'description' => __( 'Only submissions before this date (ISO8601 format).', 'jetpack-forms' ),
+							'format'      => 'date-time',
+						),
+						'after'     => array(
+							'type'        => 'string',
+							'description' => __( 'Only submissions after this date (ISO8601 format).', 'jetpack-forms' ),
+							'format'      => 'date-time',
+						),
+					),
+					'additionalProperties' => false,
+				),
+				'execute_callback'    => array( __CLASS__, 'get_form_submissions' ),
+				'permission_callback' => array( __CLASS__, 'can_edit_pages' ),
+				'meta'                => array(
+					'annotations'  => array(
+						'readonly'    => true,
+						'destructive' => false,
+						'idempotent'  => true,
+					),
+					'show_in_rest' => true,
+				),
+			)
+		);
+	}
+
+	/**
+	 * Register ability to update a form submission.
+	 *
+	 * @return void
+	 */
+	private static function register_update_submission_ability() {
+		wp_register_ability(
+			'jetpack-forms/update-submission',
+			array(
+				'label'               => __( 'Update Form Submission', 'jetpack-forms' ),
+				'description'         => __( 'Modify a form submission. Use to mark as spam, move to trash, restore from trash, or toggle read/unread state.', 'jetpack-forms' ),
+				'category'            => self::CATEGORY_SLUG,
+				'input_schema'        => array(
+					'type'                 => 'object',
+					'required'             => array( 'id' ),
+					'properties'           => array(
+						'id'        => array(
+							'type'        => 'integer',
+							'description' => __( 'The submission ID to update.', 'jetpack-forms' ),
+						),
+						'status'    => array(
+							'type'        => 'string',
+							'description' => __( 'New status: "publish" (restore), "spam" (mark spam), "trash" (soft delete).', 'jetpack-forms' ),
+							'enum'        => array( 'publish', 'draft', 'spam', 'trash' ),
+						),
+						'is_unread' => array(
+							'type'        => 'boolean',
+							'description' => __( 'Set false to mark as read, true to mark as unread.', 'jetpack-forms' ),
+						),
+					),
+					'additionalProperties' => false,
+				),
+				'execute_callback'    => array( __CLASS__, 'update_form_submission' ),
+				'permission_callback' => array( __CLASS__, 'can_edit_pages' ),
+				'meta'                => array(
+					'annotations'  => array(
+						'readonly'    => false,
+						'destructive' => false,
+						'idempotent'  => true,
+					),
+					'show_in_rest' => true,
+				),
+			)
+		);
+	}
+
+	/**
+	 * Register ability to delete a form submission.
+	 *
+	 * @return void
+	 */
+	private static function register_delete_submission_ability() {
+		wp_register_ability(
+			'jetpack-forms/delete-submission',
```

Comment:
Core still did not shipped destructive abilities, and for 7.0 there are no plans to include destructive ones either. Are we confortable in shipping some destructive abilities in Jetpack?

Reply:
I deleted it to unblock the merge. We can easily add it back in a follow-up PR once we’re confident. (I think it should be fine)